firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/HAL/Mock/I2CDevice.h

```diff
@@ -48,12 +48,23 @@ class I2CDevice : public Interfaces::I2CDevice {
    */
   I2CDeviceStatus read(uint8_t *buf, size_t count) override;
 
+  /**
+   * Reads data from the device
+   * This method should be used when data from a specific memory address needs to be read
+   * @param address of the specific register
+   * @param buf[out]    output of the data
+   * @param count   the number of bytes to be read
+   * @return ok on success, error code otherwise
+   */
```

Comment:
We need a TODO comment to implement this method and warn others that it's still unimplemented